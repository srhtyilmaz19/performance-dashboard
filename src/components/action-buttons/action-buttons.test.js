import React from "react";

import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ActionButtons from "./action-buttons";

afterEach(cleanup);

describe("<ActionButtons />", () => {
  const mockOnClick = jest.fn();

  it("should render action-buttons", function () {
    const { asFragment } = render(<ActionButtons onClick={mockOnClick} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("filter button should have class and text", function () {
    const { getByTestId } = render(<ActionButtons onClick={mockOnClick} />);

    const filterButton = getByTestId("filter-button");

    expect(filterButton).toHaveClass("button");
    expect(filterButton).toHaveTextContent("FILTER");
  });

  it("reset button should have class and text", function () {
    const { getByTestId } = render(<ActionButtons onClick={mockOnClick} />);

    const resetButton = getByTestId("reset-button");

    expect(resetButton).toHaveClass("button");
    expect(resetButton).toHaveTextContent("RESET");
  });
});
