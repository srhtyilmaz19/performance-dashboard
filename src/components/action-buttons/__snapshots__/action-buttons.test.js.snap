// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButtons /> should render action-buttons 1`] = `
<DocumentFragment>
  <button
    class="button"
    data-testid="filter-button"
  >
    FILTER
  </button>
  <button
    class="button"
    data-testid="reset-button"
  >
    RESET
  </button>
</DocumentFragment>
`;
